tools/oca_publish_modules.py

```diff
@@ -0,0 +1,97 @@
+import os
+import re
+from time import sleep
+from getpass import getpass
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.chrome.options import Options
+
+
+def main(argv):
+    repositories = argv[1:]
+    if not repositories:
+        print(
+            'Please specify repository(-ies):\n',
+            os.path.basename(__file__),
+            ' repo-name-1#branch [repo-name-2#branch, ..., repo-name-N#branch]',
+        )
+        return
+
+    chrome = os.getenv('CHROMEDRIVER')
+    if chrome == None:
+        print(
+            'Please specify CHROMEDRIVER environment variable pointing to'
+            ' valid Google Chrome Driver executable file and ensure its'
+            ' location is available via PATH environment variable.\n\n'
+            'See http://chromedriver.chromium.org/downloads'
+        )
+        return
+
+    login = input('Odoo.com publisher account:')
+    password = getpass(prompt='Odoo.com account password:')
+
+    for repository in repositories:
+        publish_repository(chrome, login, password, repository)
+
+def publish_repository(chrome, login, password, repository):
+    options = Options()
+    options.headless = True
+
+    driver = webdriver.Chrome(
+        executable_path=chrome,
+        options=options,
+    )
+    wait = WebDriverWait(driver, 10)
+
+    login(driver, login, password)
+
+    try:
+        scan_repository(driver, repository)
+    except:
+        register_repository(driver, repository)
+        scan_repository(driver, repository)
+
+def login(driver, login, password):
+    driver.get('https://www.odoo.com/web/login?redirect=%2Foauth2%2Fauth%2F%3Fscope%3Duserinfo%26redirect_uri%3Dhttps%253A%252F%252Fapps.odoo.com%252Fauth_oauth%252Fsignin%26state%3D%257B%2522p%2522%253A%2B1%252C%2B%2522r%2522%253A%2B%2522%25252F%25252Fapps.odoo.com%25252Fapps%2522%252C%2B%2522d%2522%253A%2B%2522apps%2522%257D%26response_type%3Dtoken%26client_id%3Da0a30d16-6095-11e2-9c70-002590a17fd8&scope=userinfo&mode=login&redirect_hostname=https%3A%2F%2Fapps.odoo.com&login=')
+    login_field = driver.find_element_by_id('login')
+    login_field.clear()
+    login_field.send_keys(login)
+    password_field = driver.find_element_by_id('password')
+    password_field.clear()
+    password_field.send_keys(password)
+    login_button = driver.find_element_by_xpath('.//form[@action="/web/login"]//button[@type="submit"]')
+    login_button.click()
+    wait.until(lambda driver: driver.current_url == 'https://apps.odoo.com/apps')
+
+def register_repository(driver, repository):
+    driver.get('https://apps.odoo.com/apps/upload')
+    url_field = driver.find_element_by_id('url')
+    url_field.clear()
+    url.send_keys('https://github.com/OCA/' + repository)
+    submit_button = driver.find_element_by_id('apps_submit_repo_button')
+    submit_button.click()
```

Comment:
What if there is an error? (eg: timeout, the repo is already submitted, etc.)

Reply:
This method is checked when there's need, all other issues - exception and crash